src/python/pants/backend/jvm/tasks/jvm_task.py

```diff
@@ -68,3 +76,17 @@ def classpath(self, targets, classpath_prefix=None, classpath_product=None, excl
     classpath = list(classpath_prefix or ())
     classpath.extend(classpath_for_targets)
     return classpath
+
+  def preferred_jvm_distribution_for_targets(self, targets, jdk=False):
+    """Find the preferred jvm distribution for running code from the given targets."""
+    return self.preferred_jvm_distribution(self._jvm_platforms_from_targets(targets),
+                                           jdk=jdk)
+
+  def preferred_jvm_distribution(self, platforms, jdk=False):
+    return JvmPlatform.preferred_jvm_distribution(platforms, self._strict_jvm_version, jdk=jdk)
+
+  def _jvm_platforms_from_targets(self, targets):
+    # Override this to change platform lookup.
+    # This overriding will be eliminated in the next change.
```

Comment:
Would you mind expanding on what you mean by this somewhere? Either here or in a ticket maybe?

Reply:
Ah, I see.

> - Add a cut out so that junit_run can use test_platform

Re: this cutout: is the idea that the `preferred` JVM here defaults to a default "runtime" platform? If so, would it be easier to have two methods? Also, should the split be "compiletime" and "runtime", rather than something test-specific...?

Reply:
yes. The next patch introduces `runtime_platform` and switches to using that instead of test_platform